lib/model/model.go

```diff
@@ -1558,13 +1559,129 @@ func sendIndexTo(minSequence int64, conn protocol.Connection, folder string, fs
 }
 
 func (m *Model) updateLocalsFromScanning(folder string, fs []protocol.FileInfo) {
-	m.updateLocals(folder, fs)
+	m.fmut.RLock()
+	folderCfg := m.folderCfgs[folder]
+	m.fmut.RUnlock()
+
+	// if this folder is a "write/pull only" folder, then we'll have to undo any local changes
+	if folderCfg.Type == config.FolderTypeReceiveOnly {
+		m.rejectLocalChanges(folder, fs)
```

Comment:
I still strongly suggest you create a:
```
type receiveOnlyFolder type {
  sendOnlyFolder
}
```

and override the bit that handles the scanning to do reverts instead, then this whole "detach deleteDir from the struct" business goes away.